### PR TITLE
feat: copy bssid in wifi scan if set

### PR DESCRIPTION
**Pull request by sonofviking**
Wifi scans on the Esp32 did not seem to contain any BSSID's, this should fix that

### CODE REVIEW

**Review by carlescufi — DISMISSED** [review]
Thanks for the PR! Unfortunately we do not accept anonymous contributions.
See https://docs.zephyrproject.org/latest/contribute/guidelines.html#dco-sign-off

File: drivers/wifi/esp32/src/esp_wifi_drv.c
Comment (sylvioalves) [inline drivers/wifi/esp32/src/esp_wifi_drv.c:186]: I suggest removing this `if (ap_list_buffer[k].bssid) {`. It will probably drop build warnings due to comparison.

Comment (sylvioalves) [reply drivers/wifi/esp32/src/esp_wifi_drv.c:186]: Have you checked whether build completes without any warning?

**Comment by sylvioalves** [comment]
@sonofviking, you still need to update your Name and email. Check this errors:
```c
author email (sonofviking <sonofviking@protonmail.com>) needs to match one of the signed-off-by entries.
author email (sonofviking <sonofviking@protonmail.com>) does not follow the syntax: First Last <email>.
```

File: drivers/wifi/esp32/src/esp_wifi_drv.c
Comment (sonofviking) [reply drivers/wifi/esp32/src/esp_wifi_drv.c]: I have made the check more explicit. The checkpatch script does not seem to give out any warnings if that's what you meant.

File: drivers/wifi/esp32/src/esp_wifi_drv.c
Comment (sylvioalves) [inline drivers/wifi/esp32/src/esp_wifi_drv.c:186]: `ap_list_buffer[k].bssid` will never be NULL. Just remove the if clause and it shall work. Worst scenario is empty bssid.

**Comment by sylvioalves** [comment]
@sonofviking would you squash both commits in a single one? btw, thanks for the PR, we did miss adding this.

**Comment by kartben** [comment]
> Thanks for the PR! Unfortunately we do not accept anonymous contributions. See [docs.zephyrproject.org/latest/contribute/guidelines.html#dco-sign-off](https://docs.zephyrproject.org/latest/contribute/guidelines.html#dco-sign-off)

@carlescufi  In the future I think you shouldn't actually -1 the PR: since it's not compliant anyway, CI will take care of blocking it. Having someone actually block the PR instead of will just delay the merge with no "technical" reason, hence why I had to just dismiss your review just now. Thanks!

**Review by kartben — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**
